lxd/storage/backend_lxd.go

```diff
@@ -0,0 +1,353 @@
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) BackupInstance(i Instance, targetPath string, optimized bool, snapshots bool, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) GetInstanceUsage(i Instance) (uint64, error) {
+	return 0, ErrNotImplemented
+}
+
+func (b *lxdBackend) SetInstanceQuota(i Instance, quota uint64) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) MountInstance(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) UnmountInstance(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) GetInstanceDisk(i Instance) (string, string, error) {
+	return "", "", ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateInstanceSnapshot(i Instance, name string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) RenameInstanceSnapshot(i Instance, newName string, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) DeleteInstanceSnapshot(i Instance, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) RestoreInstanceSnapshot(i Instance, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) MountInstanceSnapshot(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) UnmountInstanceSnapshot(i Instance) (bool, error) {
+	return true, ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateImage(img api.Image, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) DeleteImage(img api.Image, op *operations.Operation) error {
+	return ErrNotImplemented
+}
+
+func (b *lxdBackend) CreateCustomVolume(vol api.StorageVolumesPost, op *operations.Operation) error {
+	project := "default"
+
+	volType, err := VolumeTypeNameToType(vol.Type)
+	if err != nil {
+		return err
+	}
+
+	// Create database entry for new storage volume.
+	err = VolumeDBCreate(b.state, b.name, vol.Name, vol.Description, vol.Type, false, vol.Config)
+	if err != nil {
+		return err
+	}
+
+	revert := true
+	defer func() {
+		if revert {
+			b.state.Cluster.StoragePoolVolumeDelete(project, vol.Name, volType, b.id)
+		}
+	}()
+
+	volID, _, err := b.state.Cluster.StoragePoolNodeVolumeGetTypeByProject(project, vol.Name, volType, b.id)
+	if err != nil {
+		return err
+	}
+
+	err = b.driver.CreateVolume(drivers.VolumeTypeCustom, volID, vol.Name, op)
+	if err != nil {
+		return err
+	}
+
+	revert = false
+	return nil
+}
+
+func (b *lxdBackend) CreateCustomVolumeFromCopy(vol api.StorageVolumesPost, op *operations.Operation) error {
+	project := "default"
+	volType, err := VolumeTypeNameToType(vol.Type)
+	if err != nil {
+		return err
+	}
+
+	// Check the supplied config and remove any fields not relevant for destination pool type.
+	volConfig, err := VolumePropertiesTranslate(vol.Config, b.driver.Name())
```

Comment:
So this we'll need to do differently, I don't want to have the supported pool config and volume config be stored outside of the driver. Adding a support pool or volume option should only require a change to the driver itself and to nothing else.

Reply:
@stgraber I approached this by adding a `removeUnknownKeys` argument to the new volume validation functions, when doing a copy of a volume the supplied config is validated with `removeUnknownKeys` set to true such that keys not recognised by the destination pool's driver are removed.